core/src/main/java/net/opentsdb/aura/metrics/core/downsample/DownSampler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of OpenTSDB.
+ * Copyright (C) 2021  Yahoo.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.opentsdb.aura.metrics.core.downsample;
+
+import java.util.Iterator;
+
+public class DownSampler {
+
+  private final int interval;
+  private int numPoints;
+  private final Aggregator aggregator;
+
+  public DownSampler(final int interval, final int numPoints, final Aggregator aggregator) {
+    this.interval = interval;
+    this.numPoints = numPoints;
+    this.aggregator = aggregator;
+  }
+
+  public void apply(final double[] rawValues) {
+    aggregator.reset();
```

Comment:
We need a check somewhere that validates we don't pass in an odd interval with the wrong source segment size. E.g. if we have a downsample of 45 minutes (could happen) then we'd want to throw an error saying that's not valid for Aura.